python/clean_export_aggregate.py

```diff
@@ -0,0 +1,53 @@
+import pandas as pd
+import sys
+from sqlalchemy import create_engine
+
+BUILD_ENGINE = sys.argv[1]
+
+template_lookup = {
+    "aggregate_cdta_2020": "CDTA",
+    "aggregate_block_2020": "CensusBlocks",
+    "aggregate_tract_2020": "CensusTracts",
+    "aggregate_councildst_2010": "CityCouncil",
+    "aggregate_commntydst_2010": "CommunityDistricts",
+    "aggregate_nta_2020": "NTA",
+}
+
+
+def read_aggregate_template(name: str):
+    base = pd.read_csv(
+        f"data/agg_template/{template_lookup[name]}.csv", dtype=str)
+    base.set_index(get_index_columns(name),
+                   inplace=True, verify_integrity=True)
+    base.drop(columns=["OBJECTID", "boro"], axis=1, inplace=True)
+    return base
+
+
+def get_index_columns(name: str):
+    index_name = {
+        "aggregate_cdta_2020": "cdta2020",
+        "aggregate_block_2020": "bctcb2020",
+        "aggregate_tract_2020": "bct2020",
+        "aggregate_councildst_2010": "councildst",
+        "aggregate_commntydst_2010": "commntydst",
+        "aggregate_nta_2020": "nta2020",
+    }
+    return index_name[name]
+
+
+if __name__ == "__main__":
+
+    table = sys.argv[2]
+
+    engine = create_engine(BUILD_ENGINE)
+    base = read_aggregate_template(table)
+    df = pd.read_sql(f"""SELECT * FROM {table}""", con=engine)
+
+    df.dropna(axis=0, subset=[get_index_columns(table)], inplace=True)
+    df.set_index(get_index_columns(table), inplace=True, verify_integrity=True)
+
+    df_concat = pd.concat([base, df], axis=1)
+    final = df_concat.loc[:, ~df_concat.columns.duplicated()].copy()
```

Comment:
I googled this and I think found the same [blog post](https://towardsdatascience.com/python-pandas-dataframe-join-merge-and-concatenate-84985c29ef78) you did lol. I think there is probably a better solution upstream, working on this now

Reply:
Sorry maybe I don't understand. Can't the `base` here be just an index with no data? Why does it have any columns at all?

Reply:
Where do we use nta names? Can't use just use the `nta2020` column?

Reply:
good q. GIS when they publish it will need to include the nta names. I think this might be a preference from housing.

Reply:
So is the issue two columns of nta names? Is it possible they have different data? If so this would just keep whichever is first which is probably not best. Should probably explicitly deal with this upstream right? Unless I'm misunderstanding

Reply:
yes two columns of nta names. We do know that we should always use the first one which come from the base template instead of the one from aggregate table which likely to be incomplete. @SashaWeinstein

Reply:
Ok so let's drop the column from the aggregate table? Even if it's more code it seems cleaner

Reply:
This is my one outstanding comment before approval, think we should be more explicit about dropping the duplicate columns